backend/handlers/handlers.go

```diff
@@ -1438,6 +1439,33 @@ func (handlers *Handlers) getExchangeMoonpayBuyInfo(r *http.Request) (interface{
 	return resp, nil
 }
 
+func (handlers *Handlers) getExchangeBtcDirectInfo(r *http.Request) interface{} {
+	res := struct {
+		Success bool    `json:"success"`
+		Url     string  `json:"url"`
+		ApiKey  string  `json:"apiKey"`
+		Address *string `json:"address"`
+	}{}
+
+	code := accountsTypes.Code(mux.Vars(r)["code"])
+	acct, err := handlers.backend.GetAccountFromCode(code)
+	accountValid := acct != nil && acct.Offline() == nil && !acct.FatalError()
+	if err != nil || !accountValid {
+		res.Success = false
```

Comment:
Imho readability would be a bit better to define the struct only above, and instantiate locally, like `return &result{.. }`, like in this example: https://github.com/BitBoxSwiss/bitbox-wallet-app/blob/eb7ec055d00179a9e13afdc148b0ac6aaf9463b4/backend/handlers/handlers.go#L1577

Just a minor nit though.